packages/apollo-gateway/src/loadServicesFromStorage.ts

```diff
@@ -87,7 +87,7 @@ export async function getServiceDefinitionsFromStorage({
   } = await fetchLinkFile(baseUrl);
 
   // If the link file is a cache hit, no further work is needed
-  if (linkFileCacheHit) return { isNewSchema: false };
+  if (linkFileCacheHit) return { isNewSchema: false, serviceDefinitions: [] };
```

Comment:
hmm... this return is a little wonky. Do we have the old `serviceDefinitions` here or can we make this field undefinable?

Reply:
In fact, I think this might cause a bug, since we're now only early-returning on whether each `serviceDefinition` matches.